unison-src/transcripts/error-messages.output.md

```diff
@@ -343,10 +343,12 @@ use.keyword.in.namespace = 1
 
   Loading changes detected in scratch.u.
 
-  The identifier used here isn't allowed to be a reserved keyword: 
+  The identifier namespace used here is a reserved keyword: 
```

Comment:
Hmm I like the idea of including the text here so we don't just rely on colors. But I could see this getting confusing if it doesn't include quotes or backticks or something around the identifier.

Reply:
Ah right. I had originally hoped to modify `style` to wrap `Code` in something, but it requires not-insignificant changes to support anything other than ANSI colors there.

It’s probably good English to change

> The identifier namespace used here is a reserved keyword:

to

> The identifier, namespace, used here is a reserved keyword:

And I maybe defining a function like
```haskell
styleCode code = "‘" <> style Code code <> "’"
```
to use in the interim would be good enough? (It avoids hardcoding the quotes everywhere, but also doesn’t require generalizing all the `ColorText` stuff.)

> The identifier, ‘namespace’, used here is a reserved keyword:

Reply:
I don't feel qualified to answer that. @aryairani or @mitchellwrosen WDYT?

Reply:
I'm not familiar enough to know about generalizing `ColorText` / `SyntaxText` / whatever to support a different solution, but that `styleCode` combinator does like fine and good.

Reply:
I guess I would use backticks if anything, like in `backticked`. 

Also we have other places where we rely on color that gets lost in transcript/non-ansi output, and I'm not sure what to do about it. :-\  Once upon a time, we scrapped having separate renderers depending on the terminal mode, but long term we could look at that again.

Reply:
> I guess I would use backticks if anything, like in `backticked`.

Done. The reason I prefer Unicode quotes is because you often need to distinguish between literal backticks in the code vs backticks used to surround code. E.g., in this particular error message, it says “You can avoid this problem […] by […] wrapping it in backticks (like `namespace` ).” where we want those backticks to be considered part of the code (and they are colorized as such in contexts that allow it). But this is exactly why we have these style abstractions. If you decide to switch the way it’s quoted down the road, you just change `quoteCode` and not dozens of string literals.

> Also we have other places where we rely on color that gets lost in transcript/non-ansi output, and I'm not sure what to do about it. :-\ Once upon a time, we scrapped having separate renderers depending on the terminal mode, but long term we could look at that again.

Yeah, it’s tough. Apple’s pasteboard lets you store multiple versions of the content depending on where you paste to, but that doesn’t work (at least not easily) from the terminal, and it also relies on the receiving app having pretty sophisticated pasteboard handling. I think you mostly need to accept that anything other than text is going to be tough. Another approach could be to use ANSI escapes to hide the quotes in the terminal, so that copying the text still results in them being in the clipboard. But, yeah, a conversation for another time.

Reply:
Oh, interesting.